Add VSCode extension recommendations to redwoodjs/redwood

thedavidprice opened the pull request:
Closes #99 

Copied the rec's from CRA App. What do you gents think about this list for local dev -- anything to add or remove?

cannikin commented:
I bring this up every time but I don't think we should recommended stuff like `gitlens` which doesn't actually do anything to assist with Redwood development, it's just something that a particular developer might like or not. (Personally I don't have it because it makes the line too busy and I only care about who wrote a line like once a year and I just do a `git blame` when needed.)

I think we should only be recommending extensions that provide an actual benefit to Redwood-specific development, like prisma. I'd go so far as to say we shouldn't even include formatting plugins—just because the Redwood core team likes not having semicolons doesn't mean I should *by default* have my personal coding style automatically changed when I open a Redwood project in VSCode.

mojombo commented:
I agree on not needing to specify Git plugins, but I think we should def have a default opinion about formatting and syntax. Many people won’t care what the chosen formatting is, and for all of them, it will be very nice to never have to think about it, and for it to be consistent across Redwood apps. We should also make it easy for users to override those defaults if they have a different set of rules they prefer.

thedavidprice commented:
### Coding Style
Since this is for contributing to Redwood packages, shouldn’t we always enforce a coding style? Our PR CI check includes a linter, which uses our settings and will throw an error if code styles aren’t correct.

For Apps, I believe it is currently possible to override default style settings. See my suggested [implementation in this comment](https://github.com/redwoodjs/redwood/issues/195#issuecomment-606857872).

### Styling Extensions to Keep (y/n/maybe)
Seems like these are the ones to keep.
- dbaeumer.vscode-eslint
- mgmcdermott.vscode-language-babel
- editorconfig.editorconfig

### And these can be removed (y/n/maybe)
- pflannery.vscode-versionlens
- wix.vscode-import-cost

### Tech Specific
Are these actually helpful within package development?
- ofhumanbondage.react-proptypes-intellisense
- prisma.prisma

cannikin commented:
OH GEEZ just realized this was a recommended list for the Redwood codebase itself (which affects us), not create-redwood-app (which would affect everyone who creates a Redwood app).

I still stand by my hatred of gitlens, but I take back what I said about formatting—we should totally have a consistent style throughout Redwood itself!

From @thedavidprice list above I think we keep the first group, remove the second group, remove the third group, since we're not actually creating prisma schemas or components in Redwood proper.

thedavidprice commented:
Thanks, RC! And +1

_Going once..._

peterp commented:
+1

thedavidprice commented on .vscode/extensions.json:
```diff
@@ -0,0 +1,13 @@
+    "prisma.prisma"
+  ],
+  "unwantedRecommendations": []
+}
```
Not yet. I was waiting on #363 to see if we needed editorconfig. Have a plan today to get things moving again.

thedavidprice commented:
Cleaned up the extension recommendation list per conversation.

I also removed `.editorconfig` — unless we end up needing this to sort out Windows line ending issues, we don’t actually want to provide a way to override default config.

That leaves these two recommendations:
- dbaeumer.vscode-eslint
- mgmcdermott.vscode-language-babel
